include/swift/Option/Options.td

```diff
@@ -312,6 +312,14 @@ def enable_library_evolution : Flag<["-"], "enable-library-evolution">,
   Flags<[FrontendOption, ModuleInterfaceOption]>,
   HelpText<"Build the module to allow binary-compatible library evolution">;
 
+def explicit_intro_version : Flag<["-"], "explicit-intro-version">,
```

Comment:
I'd suggest sticking to the term `availability` rather than "intro version" (even though "availability" also covers other things), and maybe adding a verb like `check` or `require`. Driver options are forever, so it's important to pick names carefully!

Reply:
Good idea, what about `-require-explicit-availability` or `-check-public-availability`?

Reply:
Either of those sound good to me! I guess I'd lean slightly towards the first.